src/app/src/main/kotlin/org/imperial/mrc/hint/exceptions/ExceptionHandler.kt

```diff
@@ -18,21 +19,22 @@ import javax.validation.ConstraintViolationException
 
 @Order(Ordered.HIGHEST_PRECEDENCE)
 @ControllerAdvice
-class HintExceptionHandler : ResponseEntityExceptionHandler() {
+class HintExceptionHandler(private val errorCodeGenerator: ErrorCodeGenerator,
+                           private val appProperties: AppProperties) : ResponseEntityExceptionHandler() {
 
     override fun handleExceptionInternal(e: java.lang.Exception,
                                          @Nullable body: Any?,
                                          headers: HttpHeaders,
                                          status: HttpStatus,
                                          request: WebRequest): ResponseEntity<Any> {
         logger.error(e.message)
-        return ErrorDetail(status, e.message ?: "Something went wrong").toResponseEntity()
+        return ErrorDetail(status, (e.message ?: "Something went wrong").appendErrorCodeInstructions()).toResponseEntity()
     }
 
     @ExceptionHandler(HintException::class)
     protected fun handleHintException(e: HintException): ResponseEntity<Any> {
         logger.error(e.message)
-        return ErrorDetail(e.httpStatus, e.message!!).toResponseEntity()
+        return ErrorDetail(e.httpStatus, e.message!!.appendErrorCodeInstructions()).toResponseEntity()
```

Comment:
I don't think these want the error code instructions appended as these are known errors that contain their own instructions - e.g. "You must upload a shape file before uploading survey or programme data".

Reply:
I think that might be the only case where that's true! (And it shouldn't be possible to hit it in the app anyway.) The messages for TokenException and SessionException are fairly cryptic. None of these exceptions should really happen (it's not like uploading a bad file, which is a 400), so I guess the point is more the error code than the instructions, so we can find the place in the log where they happened.

Reply:
But these are all known exceptions - i.e. we throw them manually. If they're cryptic, we should improve the error messages.